experimental/test_parallel_compose.cpp

```diff
@@ -265,6 +265,100 @@ void testEpsilon() {
     assert(equal(
       gtn::detail::dataparallel::compose(g1, g2), expected));
   }
+
+  {
+    // Simple test case for input epsilon on second graph
+    Graph g1;
+    g1.addNode(true);
+    g1.addNode(false, true);
+    g1.addArc(0, 1, 1, 2);
+
+    Graph g2;
+    g2.addNode(true);
+    g2.addNode(false, true);
+    g2.addArc(0, 1, 2, 3);
+    g2.addArc(1, 1, epsilon, 0, 2.0);
```

Comment:
@shubho  This is the case that fails. `g2`s last node has an epsilon that we should explore at the state (1,1) despite the fact that `g1` does not have any arcs out of node 1.